js/widgets/page.dialog.js

```diff
@@ -113,35 +138,58 @@ return $.widget( "mobile.page", $.mobile.page, {
 		}
 	},
 
-	_setCloseBtn: function( location, text ) {
-		var dst,
-			button = this._headerCloseButton;
+	_handleButtonClick: function() {
+
+		// Render the close button unclickable after one click
+		if ( this._ui.button ) {
+			this._ui.button
+				.css( "pointer-events", "none" )
+				.attr( "tabindex", -1 );
+		}
```

Comment:
your doing nearly the exact same thing above lets make this another `toggle` method and pass a boolean.